domain/src/main/kotlin/no/nav/su/se/bakover/domain/klage/KlageTilAttestering.kt

```diff
@@ -15,30 +15,43 @@ import java.time.LocalDate
 import java.util.UUID
 import kotlin.reflect.KClass
 
-data class KlageTilAttestering private constructor(
-    override val id: UUID,
-    override val opprettet: Tidspunkt,
-    override val sakId: UUID,
-    override val saksnummer: Saksnummer,
-    override val fnr: Fnr,
-    override val journalpostId: JournalpostId,
-    override val oppgaveId: OppgaveId,
-    override val saksbehandler: NavIdentBruker.Saksbehandler,
-    override val datoKlageMottatt: LocalDate,
-    val attesteringer: Attesteringshistorikk,
-    val vilkårsvurderinger: VilkårsvurderingerTilKlage.Utfylt,
-    val vurderinger: VurderingerTilKlage.Utfylt,
-) : Klage() {
+sealed class KlageTilAttestering : Klage {
+    abstract override val id: UUID
+    abstract override val opprettet: Tidspunkt
+    abstract override val sakId: UUID
+    abstract override val saksnummer: Saksnummer
+    abstract override val fnr: Fnr
+    abstract override val journalpostId: JournalpostId
+    abstract override val oppgaveId: OppgaveId
+    abstract override val saksbehandler: NavIdentBruker.Saksbehandler
+    abstract override val datoKlageMottatt: LocalDate
+    abstract val attesteringer: Attesteringshistorikk
+    abstract val vilkårsvurderinger: VilkårsvurderingerTilKlage.Utfylt
+    abstract val vurderinger: VurderingerTilKlage?
 
-    override fun underkjenn(
-        underkjentAttestering: Attestering.Underkjent,
-        opprettOppgave: () -> Either<KunneIkkeUnderkjenne.KunneIkkeOppretteOppgave, OppgaveId>,
-    ): Either<KunneIkkeUnderkjenne, VurdertKlage.Bekreftet> {
-        if (underkjentAttestering.attestant.navIdent == saksbehandler.navIdent) {
-            return KunneIkkeUnderkjenne.AttestantOgSaksbehandlerKanIkkeVæreSammePerson.left()
-        }
-        return opprettOppgave().map { oppgaveId ->
-            VurdertKlage.Bekreftet.create(
+    data class Avvist private constructor(
+        override val id: UUID,
+        override val opprettet: Tidspunkt,
+        override val sakId: UUID,
+        override val saksnummer: Saksnummer,
+        override val fnr: Fnr,
+        override val journalpostId: JournalpostId,
+        override val oppgaveId: OppgaveId,
+        override val saksbehandler: NavIdentBruker.Saksbehandler,
+        override val datoKlageMottatt: LocalDate,
+        override val attesteringer: Attesteringshistorikk,
+        override val vilkårsvurderinger: VilkårsvurderingerTilKlage.Utfylt,
+        override val vurderinger: VurderingerTilKlage?,
```

Comment:
delme

Reply:
Class 'Avvist' is not abstract and does not implement abstract base class member public abstract val vurderinger: VurderingerTilKlage? defined in no.nav.su.se.bakover.domain.klage.KlageTilAttestering